returnn/datasets/multi_proc.py

```diff
@@ -298,6 +316,11 @@ def _get(seq_idx: int) -> Optional[DatasetSeq]:
                     except NotImplementedError as exc:
                         all_tags = NotImplementedError(f"{exc} in {dataset}")
                     parent_conn.send(("all_tags", all_tags))
+                elif msg == "get_data_keys_and_dtypes":
```

Comment:
Why do you have made this a combined getter? This looks a bit weird to me.

Reply:
Also, I don't really see any advantage of this, but then, there are several disadvantages:
* it is inconsistent to the remaining code
* there could be cases where you only need one of them but not both, and the other one would add maybe some non-trivial overhead. e.g. maybe get_data_keys is still efficient for some dataset, but now it will also trigger get_data_dtype which is slow

Reply:
I made this a combined getter because you need to call one to compute the other, and then it made more sense to me to always do both at the same time. But I can also split this up since we usually cache the information returned from the getters, which means calling `get_data_keys` twice should usually be not much slower than calling it once.